test/DynamoCoreTests/Performance/PerformanceTests.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Dynamo.Events;
+using Dynamo.Graph.Nodes;
+using Dynamo.Graph.Nodes.CustomNodes;
+using Dynamo.Graph.Workspaces;
+using NUnit.Framework;
+
+namespace Dynamo.Tests
+{
+    [TestFixture, Category("Performance")]
+    public class PerformanceTests : DynamoModelTestBase
+    {
+        private TimeSpan lastExecutionDuration = new TimeSpan();
+
+        protected override void GetLibrariesToPreload(List<string> libraries)
+        {
+            libraries.Add("VMDataBridge.dll");
+            libraries.Add("ProtoGeometry.dll");
+            libraries.Add("DesignScriptBuiltin.dll");
+            libraries.Add("DSCoreNodes.dll");
+            libraries.Add("GeometryColor.dll");
+            base.GetLibrariesToPreload(libraries);
+        }
+        public object[] FindWorkspaces()
+        {
+            var di = new DirectoryInfo(Path.Combine(TestDirectory, "core", "performance"));
+            var fis = di.GetFiles("*.dyn", SearchOption.AllDirectories);
+            return fis.Select(fi => fi.FullName).ToArray();
+        }
+
+        [TestFixtureSetUp]
+        public void FixtureSetup()
+        {
+            ExecutionEvents.GraphPostExecution += ExecutionEvents_GraphPostExecution;
+        }
+
+        [TestFixtureTearDown]
+        public void TearDown()
+        {
+            ExecutionEvents.GraphPostExecution -= ExecutionEvents_GraphPostExecution;
+        }
+
+        private void ExecutionEvents_GraphPostExecution(Session.IExecutionSession session)
+        {
+            lastExecutionDuration = (TimeSpan)session.GetParameterValue(Session.ParameterKeys.LastExecutionDuration);
```

Comment:
@sm6srw did you intend to use this `lastExecutionDuration` timespan somewhere?

Reply:
Yes, but I forgot about it.

Reply:
Removed the timer for now.